addons/payment_payulatam/models/payment.py

```diff
@@ -37,8 +38,9 @@ def _payulatam_generate_sign(self, inout, values):
             data_string = ('~').join((self.payulatam_api_key, self.payulatam_merchant_id, values['referenceCode'],
                                       str(values['amount']), values['currency']))
         else:
+            rounded_amount = decimal.Decimal(values.get('TX_VALUE')).quantize(decimal.Decimal('0.1'), decimal.ROUND_HALF_EVEN)
             data_string = ('~').join((self.payulatam_api_key, self.payulatam_merchant_id, values['referenceCode'],
-                                      str(float(values.get('TX_VALUE'))), values['currency'], values.get('transactionState')))
+                                      str(rounded_amount), values['currency'], values.get('transactionState')))
```

Comment:
We have methods currency.round or float_repr or float_round to handle these cases.

Reply:
@jco-odoo Do they implement [rounding half to even](https://en.wikipedia.org/wiki/Rounding#Round_half_to_even)?